### PR TITLE
Fix downloading huge files (broken since hyperup)

### PR DESCRIPTION
**Pull request by Eijebong**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22144)

### CODE REVIEW

**Comment by Eijebong** [comment]
cc @Manishearth

**Comment by Eijebong** [comment]
`[Errno 12] Cannot allocate memory`

File: components/net/connector.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ impl Stream for WrappedBody {
                         Decoder::Plain => Some(chunk),
                         Decoder::Gzip(Some(ref mut decoder)) => {
                             let mut buf = vec![0; BUF_SIZE];
-                            *decoder.get_mut() = Cursor::new(chunk.into_bytes());
+                            decoder.get_mut().get_mut().extend(&chunk.into_bytes());
```
Comment (nox) [inline components/net/connector.rs]: `&chunk.into_bytes()` should be `chunk.as_ref()`.

Comment (Eijebong) [reply components/net/connector.rs]: Done

**Pull request merged**